pypeit/edgetrace.py

```diff
@@ -4855,13 +4863,17 @@ def order_refine(self, debug=False):
             add_left, add_right = self.order_refine_fixed_format(reference_row, debug=debug)
             rmtraces = None
         else:
+            # TODO: `bracket` is hard-coded!  Currently I expect we always want
+            # to set bracket=True, but we should plan to revisit this and maybe
+            # expose as a user parameter.
+            bracket = True
             add_left, add_right, rmtraces \
-                    = self.order_refine_free_format(reference_row, debug=debug)
+                    = self.order_refine_free_format(reference_row, bracket=bracket, debug=debug)
 
         if add_left is None or add_right is None:
             msgs.info('No additional orders found to add')
             return
-
+        
```

Comment:
Hey @kbwestfall , I don't know if this is the best way, but by adding 
`self.clean_traces(force_flag='OFFDETECTOR', rebuild_pca=False, sync_mode='both', assume_synced=True)`
below, after `self.insert`, it fixed the problem.
I have noticed that in the case of added orders we need to force the flag `OFFDETECTOR`, because those orders have also the flag `ORDERINSERT` therefore 
`self.fully_masked_traces(flag=self.bitmask.bad_flags, exclude=self.bitmask.exclude_flags)`
alone will exclude those orders.

Reply:
The specific cause of the added orders was because the code always added the "bracketing" orders, but it only removed them if the code detected overlap.  I updated the code so that it now also removes those bracketing orders when there is no overlap, as it should have done to begin with!  Sorry for these bugs.

This means that we shouldn't need to run `clean_traces`.